extensions/amp-bind/amp-bind.md

```diff
@@ -66,425 +72,1074 @@ The `amp-bind` component allows you to add custom stateful interactivity to your
+- The `[text]` value contains the **expression**, `'Hello ' + foo`.
+  - This expression concatenates the string 'Hello ' and the value of the **state variable** foo.
+- When the user taps/clicks the button, it triggers the `tap` event.
+- The `tap` event invokes the `AMP.setState()` method.
+- The `AMP.setState()` methods sets the `foo` **state variable** to the value of `Interactivity`.
+- The sate is no longer empty, so the pages updates the bound property to its state.
 
-`amp-bind` takes special care to ensure speed, security and performance on AMP pages.
+[tip type="note"]
+Calling `AMP.setState()` in some examples may set or change states of other examples on page. Refresh this page to see examples before `AMP.setState()`.
+[/tip]
 
 ### A slightly more complex example
 
 [filter formats="websites, stories, ads"]
 
+[example preview="top-frame" playground="true" imports="amp-bind"]
+
 ```html
-<!-- Store complex nested JSON data in <amp-state> elements. -->
-<amp-state id="myAnimals">
+  <style amp-custom>
+    .greenBorder {
+      border: 5px solid green;
+    }
+    .redBorder {
+      border: 5px solid red;
+    }
+  </style>
+</head>
+<body>
+<amp-state id="theFood">
   <script type="application/json">
     {
-      "dog": {
-        "imageUrl": "/img/dog.jpg",
-        "style": "greenBackground"
+      "cupcakes": {
+        "imageUrl": "https://amp.dev/static/samples/img/image2.jpg",
+        "style": "greenBorder"
       },
-      "cat": {
-        "imageUrl": "/img/cat.jpg",
-        "style": "redBackground"
+      "sushi": {
+        "imageUrl": "https://amp.dev/static/samples/img/image3.jpg",
+        "style": "redBorder"
       }
     }
   </script>
 </amp-state>
+<div class="greenBorder" [class]="theFood[currentMeal].style">
+  <p>Each food has a different border color.</p>
+  <p [text]="'I want to eat ' + currentMeal + '.'">I want to eat cupcakes.</p>
+  <amp-img
+  width="300"
+  height="200"
+  src="https://amp.dev/static/samples/img/image2.jpg"
+  [src]="theFood[currentMeal].imageUrl">
+  </amp-img>
+  <button on="tap:AMP.setState({currentMeal: 'sushi'})">Set to sushi</button>
+  <button on="tap:AMP.setState({currentMeal: 'cupcakes'})">Set to cupcakes</button>
+</div>
+```
 
-<p [text]="'This is a ' + currentAnimal + '.'">This is a dog.</p>
+[/example]
 
-<!-- CSS classes can also be added or removed with [class]. -->
-<p class="greenBackground" [class]="myAnimals[currentAnimal].style">
-  Each animal has a different background color.
-</p>
+In the example above:
 
-<!-- Or change an image's src with the [src] binding. -->
-<amp-img
-  width="300"
-  height="200"
-  src="/img/dog.jpg"
-  [src]="myAnimals[currentAnimal].imageUrl"
->
-</amp-img>
+- The `<amp-state>` component holds the complex nested JSON data. It has an `id` of "theFood" to allow us to reference the defined data. But because `<amp-bind>` does not evaluate `<amp-state>` on page load, the **state** is empty.
```

Comment:
`the complex nested JSON data` sounds a bit weird, maybe: `declares state using a JSON string literal.`